src/Compilers/Core/Portable/Text/StringTextWriter.cs

```diff
@@ -35,6 +30,7 @@ public override Encoding Encoding
 
         public override SourceText ToSourceText()
         {
+            RoslynDebug.Assert(_builder.Length == _length);
```

Comment:
this is interesting.  is it known that in all cases StringTextWriter is using the up-front provided length as the total-length of the result?  Not just as a capacity hint on how much space to initially allocate?

Reply:
@CyrusNajmabadi Yes, I checked all the call sites.

Reply:
can you doc this on the type/constructor then?  thanks!

Reply:
Done.